onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -1684,24 +1682,45 @@ TEST_F(GraphTransformationTests, GeluFusionTestFormat2GraphInput) {
   ASSERT_TRUE(op_to_count["Gelu"] == 1);
 }
 
-TEST_F(GraphTransformationTests, BiasGeluTest) {
-  auto model_uri = MODEL_FOLDER "fusion/bias_gelu_fusion.onnx";
+namespace {
+void TestBiasGeluFusion(const PathString& model_uri, const Logger& logger) {
   std::shared_ptr<Model> p_model;
-  ASSERT_STATUS_OK(Model::Load(model_uri, p_model, nullptr, *logger_));
+  ASSERT_STATUS_OK(Model::Load(model_uri, p_model, nullptr, logger));
   Graph& graph = p_model->MainGraph();
 
   onnxruntime::GraphTransformerManager graph_transformation_mgr{5};
   graph_transformation_mgr.Register(onnxruntime::make_unique<GeluFusion>(), TransformerLevel::Level2);
-  graph_transformation_mgr.Register(onnxruntime::make_unique<BiasGelu>(), TransformerLevel::Level2);
-  auto ret = graph_transformation_mgr.ApplyTransformers(graph, TransformerLevel::Level2, *logger_);
-  ASSERT_TRUE(ret.IsOK());
+  graph_transformation_mgr.Register(onnxruntime::make_unique<BiasGeluFusion>(), TransformerLevel::Level2);
+  ASSERT_STATUS_OK(graph_transformation_mgr.ApplyTransformers(graph, TransformerLevel::Level2, logger));
   std::map<std::string, int> op_to_count = CountOpsInGraph(graph);
-  ASSERT_TRUE(op_to_count["Div"] == 0);
-  ASSERT_TRUE(op_to_count["Add"] == 0);
-  ASSERT_TRUE(op_to_count["Erf"] == 0);
-  ASSERT_TRUE(op_to_count["Mul"] == 0);
-  ASSERT_TRUE(op_to_count["Gelu"] == 0);
-  ASSERT_TRUE(op_to_count["BiasGelu"] == 1);
+  ASSERT_EQ(op_to_count["Div"], 0);
+  ASSERT_EQ(op_to_count["Add"], 0);
+  ASSERT_EQ(op_to_count["Erf"], 0);
+  ASSERT_EQ(op_to_count["Mul"], 0);
+  ASSERT_EQ(op_to_count["Gelu"], 0);
+  ASSERT_EQ(op_to_count["BiasGelu"], 1);
+
+  // verify second input is bias with 1 dimension
+  auto bias_gelu_it = std::find_if(
+      graph.Nodes().begin(), graph.Nodes().end(),
+      [](Node& n) { return n.OpType() == "BiasGelu"; });
+  ASSERT_NE(bias_gelu_it, graph.Nodes().end());
+  const auto& bias_gelu_inputs = bias_gelu_it->InputDefs();
+  ASSERT_EQ(bias_gelu_inputs.size(), 2);
+  const auto* bias_shape = bias_gelu_inputs[1]->Shape();
+  ASSERT_NE(bias_shape, nullptr);
+  ASSERT_EQ(bias_shape->dim_size(), 1);
+}
+}  // namespace
+
+TEST_F(GraphTransformationTests, BiasGeluTest) {
+  auto model_uri = MODEL_FOLDER "fusion/bias_gelu_fusion.onnx";
+  TestBiasGeluFusion(model_uri, *logger_);
+}
+
+TEST_F(GraphTransformationTests, BiasGeluWithAddInputsSwappedTest) {
+  auto model_uri = MODEL_FOLDER "fusion/bias_gelu_fusion_add_inputs_swapped.onnx";
```

Comment:
>_add_inputs_swapped [](start = 56, length = 19)

why is this called swapped?

Reply:
it tests the case where the inputs to the fused add are [bias, input], as opposed to [input, bias]